packages/sitecore-jss-nextjs/src/utils/utils.ts

```diff
@@ -8,20 +8,24 @@ import { isEditorActive, resetEditorChromes } from '@sitecore-jss/sitecore-jss/u
  * VERCEL_URL is provided by Vercel in case if we are in Preview deployment (deployment based on the custom branch),
  * preview deployment has unique url, we don't know exact url.
  * Similarly, DEPLOY_URL is provided by Netlify and would give us the deploy URL
+ * In production non-editing environments it is desirable to use relative urls, so in that case set PUBLIC_URL = ''
  */
 export const getPublicUrl = (): string => {
-  if (process.env.NETLIFY && process.env.DEPLOY_URL) return process.env.DEPLOY_URL;
-  if (process.env.VERCEL_URL) return `https://${process.env.VERCEL_URL}`;
-
   let url = process.env.PUBLIC_URL;
+
   if (url === undefined) {
+    if (process.env.NETLIFY && process.env.DEPLOY_URL) return process.env.DEPLOY_URL;
+    if (process.env.VERCEL_URL) return `https://${process.env.VERCEL_URL}`;
+
     console.warn(
       `${chalk.yellow.bold(
         'Warning:'
       )} An PUBLIC_URL environment variable is not defined. Falling back to http://localhost:3000.`
     );
```

Comment:
I think we should get rid of this warning since we are now removing the default .env value (of http://localhost:3000) and it'll be shown in the logs with a default local setup. It's less of an issue now and more just the expected default (covered already in the JSDoc comment).